models/iaf_cond_alpha_mc_fixedca.h

```diff
@@ -0,0 +1,663 @@
+
+// Generated includes:
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// C++ includes:
+#include <vector>
+
+// C includes:
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_odeiv.h>
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "recordables_map.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+// Includes from sli:
+#include "dictdatum.h"
+#include "name.h"
+
+/* BeginDocumentation
+Name: iaf_cond_alpha_mc_fixedca - PROTOTYPE Multi-compartment conductance-based leaky
+integrate-and-fire
+neuron model with calcium spike at distal compartment modeled using threshold triggered fixed
+waveform.
+
+Description:
+THIS MODEL IS A PROTOTYPE FOR ILLUSTRATION PURPOSES. IT IS NOT YET
+FULLY TESTED. USE AT YOUR OWN PERIL!
+
+iaf_cond_alpha_mc_fixedca is an implementation of a multi-compartment spiking
+neuron using IAF dynamics with conductance-based synapses. It serves
+mainly to illustrate the implementation of multicompartment models in
+NEST.
+
+The model has three compartments: soma, proximal and distal dendrite,
+labeled as s, p, and d, respectively. Compartments are connected through
+passive conductances as follows
+
+C_m.s d/dt V_m.s = ... - g_sp ( V_m.s - V_m.p )
+
+C_m.p d/dt V_m.p = ... - g_sp ( V_m.p - V_m.s ) - g_pd ( V_m.p - V_m.d )
+
+C_m.d d/dt V_m.d = ...                          - g_pd ( V_m.d - V_m.p ) + ICa
+
+A spike is fired when the somatic membrane potential exceeds threshold,
+V_m.s >= V_th. Upon threshold crossing, somatic membrane potential is set to
+a maximum potential, V_m.s == V_max, and somatic leak set to a larger leak value
+for the refractory period. Depending on value of the reset flag, the somatic
+membrane potential maybe reset to reset value at end of refractory period.
+To emulate backpropagating action potential, an alpha shaped current is
+introduced 1ms and 2ms after spike at proximal and distal compartment respectively.
+Dendritic membrane potentials are not manipulated after a spike.
+The spike threshold is adaptive, whereby it jumps by a jump value upon crossing,
+and then decays exponentially back to baseline value.
+
+There is one excitatory and one inhibitory conductance-based synapse
+onto each compartment, with alpha-function time course. The alpha
+function is normalised such that an event of weight 1.0 results in a
+peak current of 1 nS at t = tau_syn. Each compartment can also receive
+current input from a current generator, and an external (rheobase)
+current can be set for each compartment.
+
+Synapses, including those for injection external currents, are addressed through
+the receptor types given in the receptor_types entry of the state dictionary. Note
+that in contrast to the single-compartment iaf_cond_alpha model, all synaptic
+weights must be positive numbers!
+
+Depending on value of active flag, calcium spike maybe triggered upon threshold crossing,
```

Comment:
"a calcium spike"

I am a unsure about "may be triggered". This sounds as if the spike may be triggered, or not, on threshold crossing. But I suppose the calcium spike *will* be triggered, except if the previous Ca spike is still active?